flink-core/src/main/java/org/apache/flink/api/common/operators/MailboxExecutor.java

```diff
@@ -86,6 +87,25 @@ public interface MailboxExecutor {
     /** A constant for empty args to save on object allocation. */
     Object[] EMPTY_ARGS = new Object[0];
 
+    /** Extra options to configure enqueued mails. */
+    @Experimental
+    interface MailOptions {
```

Comment:
Why is it an interface and not a class?

Reply:
Because of public api - to make it as small as possible.

Reply:
But the API of a class and API of an interface is the same, isn't it?
The number of lines of code is different, but users shouldn't be worried about it.

Besides of that, `MailboxExecutor` is `PublicEvolving` and this type is `Experimental`, so it can be changed if really necessary.

Reply:
Interface has only setters, as those are the only important bit for the user. Getters are in concrete internal class.

Reply:
I'm not sure if hiding getters makes sense (users might be confused and even need them for testing). But that can be changed later if needed.